### ISSUE TITLE
Duplicate rows in Cohort

### ISSUE THREAD

**Issue by gowthamrao** [OHDSI/Atlas#375]
[Request for Atlas UX changes for new Cohort Exit Criteria.pdf](https://github.com/OHDSI/Atlas/files/1218562/Request.for.Atlas.UX.changes.for.new.Cohort.Exit.Criteria.pdf)
In our instance, duplicate rows were produced. This probably can be resolved by using distinct as part of the SQL script. But, we dont know if this is an Atlas issue or how ETL was done into OMOP.

### Expected behavior
Rows should be unique combination of cohort_definition_id, subject_id, cohort_start_date, cohort_end_date

### Actual behavior
Duplicate rows with same data

### Steps to reproduce behavior
http://www.ohdsi.org/web/atlas/#/cohortdefinition/98643

For the above query, visit_occurrence has duplicate rows. During our ETL, we allowed visits with same dates to be different rows i.e. there are multiple rows for the same person with same visit_start_date and visit_end_date. Is this the reason?

**Comment by chrisknoll** [comment]
Yes, that's the reason, but if you are only interested in a single person in the cohort, you can specify 'use earliest event per person' and that will limit the cohort to a single value per person.

However, it does seem reasonable to select the primary events by doing:

```
select person_id, start_date, max(end_date) as end_date
FROM #qualified_events
GROUP BY person_id, start_date
```

This would ensure you get 1 person per start date, but also if you have the same start date, but different end dates, the longest of the end dates will be selected (for example, if you selected a primary event based on a drug class, and drug A and drug B were found that both started on the same day, but drug B had a longer exposure, it would use drug B's exposure as the exposure end date.

If that sounds agreeable, I can implement this.

**Comment by gowthamrao** [comment]
Thank you @chrisknoll . Your suggestion makes sense - longest of the end dates for the same person_id and start_date will be of immense help.

Please implement. 👍

**Comment by gowthamrao** [comment]
@chrisknoll  - Good morning. Thank you for making this suggestion for the feature. Is this something that has been implemented?

**Comment by chrisknoll** [comment]
No, this is still an open issue.

**Comment by gowthamrao** [comment]
Closed - Cohort era-fy option OHDSI/WebAPI#234
